### PR TITLE
Add examples for image-to-text task

### PR DESCRIPTION
**Pull request by sharma-riti**
# Description
Add examples for image-to-text task

# Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] Pull request includes test coverage for the included changes.
- [ ] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure/azureml-examples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

File: cli/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
+set -x
+# The commands in this file map to steps in this notebook: https://aka.ms/azureml-infer-batch-sdk-image-classification
```
Comment (PhaniShekhar) [inline cli/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.sh:2]: Nit: Should the link be updated to corresponding image-to-text notebook

File: cli/foundation-models/system/inference/image-to-text/prepare_data.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    sample_image_2 = os.path.join(dataset_dir, "images", "1.jpg")
+
+    request_json = {
+        "input_data": {
+            "columns": ["image"],
+            "index": [0, 1],
+            "data": [
+                [
+                    base64.encodebytes(read_image(sample_image_1)).decode("utf-8"),
+                ],  # the labels are required in the first row of the "text" column
```
Comment (PhaniShekhar) [inline cli/foundation-models/system/inference/image-to-text/prepare_data.py:76]: The comments about "text" column here are not relevant.

File: cli/foundation-models/system/inference/image-to-text/prepare_data.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    with open(request_file_name, "w") as request_file:
+        json.dump(request_json, request_file)
+
+
+def prepare_data_for_batch_inference(dataset_dir: str) -> None:
+    """Prepare image folder and csv file for batch inference.
+
+    This function will move all images to a single image folder and also create a folder of csv
+    files with images in base64 format and the candidate labels.
```
Comment (PhaniShekhar) [inline cli/foundation-models/system/inference/image-to-text/prepare_data.py:94]: Nit: Candidate labels is not relevant here.

File: sdk/python/foundation-models/system/inference/image-to-text/image-to-text-online-endpoint.ipynb [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+    ")\n",
+    "\n",
+    "# The models are available in the AzureML system registry, \"azureml\"\n",
+    "registry_ml_client = MLClient(\n",
+    "    credential,\n",
+    "    subscription_id,\n",
+    "    resource_group,\n",
+    "    registry_name=\"automl-images-dev\",\n",
+    ")\n",
+    "# Generating a unique timestamp that can be used for names and versions that need to be unique\n",
```
Comment (PhaniShekhar) [inline sdk/python/foundation-models/system/inference/image-to-text/image-to-text-online-endpoint.ipynb:83]: timestamp is updated below. Maybe we can remove this part.

File: sdk/python/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.ipynb [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 3.2 Prepare a folder of CSV files with base64 images for batch inference input\n",
+    "\n",
+    "We can provide input images to batch inference in a csv file containing a column named \"image\" having either images in base64 format or publicly accessible image URLs. We provide the candidate labels in the same csv file in the first row of a column named \"text\".\n",
```
Comment (PhaniShekhar) [inline sdk/python/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.ipynb:208]: The comments about "text" column are not relevant here.

**Review by PhaniShekhar — CHANGES_REQUESTED** [review]

File: cli/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.sh (hunk 1 above)
Comment (sharma-riti) [reply cli/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.sh]: fixed

**Review by PhaniShekhar — APPROVED** [review]

File: cli/foundation-models/system/inference/image-to-text/prepare_data.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    """
+
+    csv_folder_path = os.path.join(dataset_dir, "batch")
+    os.makedirs(csv_folder_path, exist_ok=True)
+    batch_input_file = "batch_input.csv"
+    dataset_dir = os.path.join(dataset_dir, "images")
+
+    image_list = []
+    for image in os.listdir(dataset_dir):
+        with open(os.path.join(dataset_dir, image), "rb") as f:
```
Comment (bhimar) [inline cli/foundation-models/system/inference/image-to-text/prepare_data.py:101]: nit: use read_image() function defined above

File: sdk/python/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.ipynb [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+    "workspace_ml_client = MLClient(\n",
+    "    credential, subscription_id, resource_group, workspace_name\n",
+    ")\n",
+    "\n",
+    "# The models are available in the AzureML system registry, \"azureml\"\n",
+    "registry_ml_client = MLClient(\n",
+    "    credential,\n",
+    "    subscription_id,\n",
+    "    resource_group,\n",
+    "    registry_name=\"automl-images-dev\",\n",
```
Comment (bhimar) [inline sdk/python/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.ipynb:80]: restore registry name

File: sdk/python/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.ipynb [hunk 8]
```diff
@@ -0,0 +1,477 @@
...
+    "    f\"\\n\\nUsing model name: {foundation_model.name}, version: {foundation_model.version}, id: {foundation_model.id} for inferencing\"\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Prepare data for inference - Using a folder of csv files with base64 images\n",
```
Comment (bhimar) [inline sdk/python/foundation-models/system/inference/image-to-text/image-to-text-batch-endpoint.ipynb:151]: csv -> CSV for consistency with outline at the beginning of the notebook

File: sdk/python/foundation-models/system/inference/image-to-text/image-to-text-online-endpoint.ipynb [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+    "workspace_ml_client = MLClient(\n",
+    "    credential, subscription_id, resource_group, workspace_name\n",
+    ")\n",
+    "\n",
+    "# The models are available in the AzureML system registry, \"azureml\"\n",
+    "registry_ml_client = MLClient(\n",
+    "    credential,\n",
+    "    subscription_id,\n",
+    "    resource_group,\n",
+    "    registry_name=\"automl-images-dev\",\n",
```
Comment (bhimar) [inline sdk/python/foundation-models/system/inference/image-to-text/image-to-text-online-endpoint.ipynb:81]: restore registry to azureml

File: sdk/python/foundation-models/system/inference/image-to-text/image-to-text-online-endpoint.ipynb [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+    "        return f.read()\n",
+    "\n",
+    "\n",
+    "request_json = {\n",
+    "    \"input_data\": {\n",
+    "        \"columns\": [\"image\"],\n",
+    "        \"index\": [0, 1],\n",
+    "        \"data\": [\n",
+    "            [\n",
+    "                base64.encodebytes(read_image(sample_image_1)).decode(\"utf-8\"),\n",
```
Comment (bhimar) [inline sdk/python/foundation-models/system/inference/image-to-text/image-to-text-online-endpoint.ipynb:280]: nit: use same whitespace formatting for both image samples

Comment (bhimar) [reply sdk/python/foundation-models/system/inference/image-to-text/image-to-text-online-endpoint.ipynb:280]: follow whatever black reformatter does

**Review by bhimar — APPROVED** [review]

**Pull request merged**